lib/ClangImporter/ClangImporter.cpp

```diff
@@ -2868,7 +2880,7 @@ class FilteringVisibleDeclConsumer : public swift::VisibleDeclConsumer {
 
   void foundDecl(ValueDecl *VD, DeclVisibilityKind Reason,
                  DynamicLookupInfo dynamicLookupInfo) override {
-    if (isVisibleFromModule(ModuleFilter, VD))
+    if (!VD->hasClangNode() || isVisibleFromModule(ModuleFilter, VD))
```

Comment:
Why do we need `!VD->hasClangNode()` here?

Reply:
Good question! Basically it can happen that we try looking up the pure swift function that we synthesize which doesn't have a clang node (since we add it as an altDecl to the original clang operator that we import). And if that happens it will call `isVisibleFromModule` which assume it has clangNode which our pure swift function does not have. So this guards against that.

Reply:
It's when VD isn't something we imported, but something we found via alternativeDecls. I think it would be a good idea to add a comment saying that. Maybe also add a comment to the top of the consumer that says it only filters imported decls/decls that have an associated clang node.

Reply:
Hmm, does that mean that operators are effectively visible from Swift even if their module isn't imported? E.g.
```cpp
// Foo.h
static MyClass operator+(MyClass a, MyClass b) { /* ... */ };
```
```swift
// A.swift
@_implementationOnly import Foo
```
```swift
// B.swift
import A
// + shouldn't be visible here
```

Could we instead add a clang decl to the synthesized functions to avoid a special case here? That's how this is handled for synthesized subscripts.

Reply:
@egorzhdan is that done in subscripts `result->addMember(subscript)` ?

Reply:
This happens in `SwiftDeclConverter::makeSubscript`, `SubscriptDecl *subscript` is created with a `getterImpl->getClangNode()` parameter.

Reply:
hmm.. I was using `createImplicit` which doesn't have that option. But maybe `createImported` which does would be better here? And pass in the clangNode? @zoecarver ?

Reply:
feels a bit weird just cause this function technically wasn't imported.